src/Compilers/CSharp/Test/Symbol/Symbols/CovariantReturnTests.cs

```diff
@@ -0,0 +1,278 @@
+                );
+            CreateCompilation(source, parseOptions: TestOptions.WithCovariantReturns).VerifyDiagnostics(
+                );
+        }
+
+        [Fact]
+        public void CovariantReturns_02()
+        {
+            var source = @"
+class Base
+{
+    public virtual T M<T, U>() where T : class where U : class, T => null;
+}
+class Derived : Base
+{
+    public override U M<T, U>() => null;
+}
+";
+            CreateCompilation(source, parseOptions: TestOptions.WithoutCovariantReturns).VerifyDiagnostics(
+                // (8,23): error CS8652: The feature 'covariant returns' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //     public override U M<T, U>() => null;
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "M").WithArguments("covariant returns").WithLocation(8, 23)
+                );
+            CreateCompilation(source, parseOptions: TestOptions.WithCovariantReturns).VerifyDiagnostics(
+                );
+        }
+
+        [Fact]
+        public void CovariantReturns_03()
+        {
+            var source = @"
+class Base<T> where T : class
+{
+    public virtual T M() => null;
+}
+class Derived<T, U> : Base<T> where T : class where U : class, T
+{
+    public override U M() => null;
+}
+";
+            CreateCompilation(source, parseOptions: TestOptions.WithoutCovariantReturns).VerifyDiagnostics(
+                // (8,23): error CS8652: The feature 'covariant returns' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //     public override U M() => null;
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "M").WithArguments("covariant returns").WithLocation(8, 23)
+                );
+            CreateCompilation(source, parseOptions: TestOptions.WithCovariantReturns).VerifyDiagnostics(
+                );
+        }
+
+        [Fact]
+        public void CovariantReturns_04()
+        {
+            var source = @"
+class N { }
+class Base
+{
+    public virtual N M() => null;
+}
+class Derived<T> : Base where T : N
+{
+    public override T M() => null;
+}
+";
+            CreateCompilation(source, parseOptions: TestOptions.WithoutCovariantReturns).VerifyDiagnostics(
+                // (9,23): error CS8652: The feature 'covariant returns' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //     public override T M() => null;
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "M").WithArguments("covariant returns").WithLocation(9, 23)
+                );
+            CreateCompilation(source, parseOptions: TestOptions.WithCovariantReturns).VerifyDiagnostics(
+                );
+        }
+
+        [Fact]
+        public void CovariantReturns_05()
+        {
+            var source = @"
+class Base
+{
+    public virtual object M => null;
```

Comment:
>object M => null; [](start = 19, length = 17)

I think it would be interesting to test the following scenarios:
1. an attempt to override a get/set property with a get-set property
2. an attempt to override a get/set property with a get-only property
3. similar as the second case only with next level derived type attempting to override with get/set or set-only property
4. similar as the second case only with next level derived type attempting to override with more specific get-only property

Reply:
Please add test suggestions to the test plan at [https://github.com/dotnet/roslyn/issues/43188](https://github.com/dotnet/roslyn/issues/43188).  I'm actively adding tests that respond to items documented there.  I will add these.

---
In reply to: [416175895](https://github.com/dotnet/roslyn/pull/43576#discussion_r416175895) [](ancestors = 416175895)